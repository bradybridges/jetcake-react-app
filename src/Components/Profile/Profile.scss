#profile-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  justify-content: space-between;
}

#profile-img {
  border-radius: 50%;
  width: 250px;
  height: 250px;
}

#edit-btn-container {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: transform .2s;
  img {
    width: 30px;
    height: 30px;
  }
  &:hover {
    cursor: pointer;
    transform: scale(1.1);
  }
}