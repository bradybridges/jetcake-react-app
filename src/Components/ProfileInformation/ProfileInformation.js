import React from 'react';

export default function ProfileInformation({ email, address, phone, dob, securityOne, securityTwo, securityThree }) {
  return (
    <table>
      <tbody>
        <tr>
          <td>Email</td>
          <td>{email}</td>
        </tr>
        <tr>
          <td>Address</td>
          <td>{address}</td>
        </tr>
        <tr>
          <td>Phone Number</td>
          <td>{phone}</td>
        </tr>
        <tr>
          <td>Birthday</td>
          <td>{dob}</td>
        </tr>
        <tr>
          <td>Mother's maiden name?</td>
          <td>{securityOne}</td>
        </tr>
        <tr>
          <td>Name of first pet?</td>
          <td>{securityTwo}</td>
        </tr>
        <tr>
          <td>City where you grew up?</td>
          <td>{securityThree}</td>
        </tr>
      </tbody>
    </table>
  );
}
