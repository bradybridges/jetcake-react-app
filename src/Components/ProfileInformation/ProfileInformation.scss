#profile-information-container {
  width: 100%;
  display: flex;
  justify-content: center;
}

tbody {
  width: 50%;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: blueviolet;
  color: white;
  padding: 10px;
  border-radius: 4px;
}

