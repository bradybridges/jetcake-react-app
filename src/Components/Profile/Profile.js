import React, { Component } from 'react';
import ProfileInformation from '../ProfileInformation/ProfileInformation';
import EditProfileForm from '../EditProfileForm/EditProfileForm';
const editImg = require('../../Assets/edit.png');

export default class Profile extends Component{
  state = {
    showEditProfile: false,
  }

  toggleShowEditProfile = () => {
    const updatedState = !this.state.showEditProfile;
    this.setState({ showEditProfile: updatedState });
  }

  render() {
    const {
      email, 
      address, 
      phone, 
      dob, 
      securityOne, 
      securityTwo, 
      securityThree, 
      photoURL, 
      handleUpdateProfile 
    } = this.props;

    return (
      <section id="profile-container">
        <img id="profile-img" src={photoURL} alt="profile" />
        <div 
          id="edit-btn-container" 
          style={{backgroundColor: 'black'}}
          onClick={() => this.setState({ showEditProfile: true })}
        >
          <img src={editImg} alt="edit" />
        </div>
        {this.state.showEditProfile && <EditProfileForm 
          email={email}
          address={address}
          phone={phone}
          dob={dob}
          securityOne={securityOne}
          securityTwo={securityTwo}
          securityThree={securityThree}
          toggleShowEditProfile={this.toggleShowEditProfile}
          handleUpdateProfile={handleUpdateProfile}
        />}
        { !this.state.showEditProfile && <ProfileInformation 
          email={email}
          address={address}
          phone={phone}
          dob={dob}
          securityOne={securityOne}
          securityTwo={securityTwo}
          securityThree={securityThree}
        />
        }
      </section>
    );
  }
}

