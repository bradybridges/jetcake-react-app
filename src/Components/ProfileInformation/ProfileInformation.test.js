import React from 'react';
import { shallow } from 'enzyme';
import ProfileInformation from './ProfileInformation';
import Profile from '../Profile/Profile';

describe('ProfileInformation', () => {
  const wrapper = shallow(<ProfileInformation
    email="test@test.com"
    address="123 test blvd"
    phone="1112223333"
    dob="1990-09-09"
    securityOne="Randall"
    securityTwo="Fido"
    securityThree="Minneapolis"
  />);
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
