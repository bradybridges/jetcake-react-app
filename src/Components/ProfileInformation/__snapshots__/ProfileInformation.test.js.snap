// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInformation should match snapshot 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        Email
      </td>
      <td>
        test@test.com
      </td>
    </tr>
    <tr>
      <td>
        Address
      </td>
      <td>
        123 test blvd
      </td>
    </tr>
    <tr>
      <td>
        Phone Number
      </td>
      <td>
        1112223333
      </td>
    </tr>
    <tr>
      <td>
        Birthday
      </td>
      <td>
        1990-09-09
      </td>
    </tr>
    <tr>
      <td>
        Mother's maiden name?
      </td>
      <td>
        Randall
      </td>
    </tr>
    <tr>
      <td>
        Name of first pet?
      </td>
      <td>
        Fido
      </td>
    </tr>
    <tr>
      <td>
        City where you grew up?
      </td>
      <td>
        Minneapolis
      </td>
    </tr>
  </tbody>
</table>
`;
